test/jdk/java/awt/datatransfer/Independence/IndependenceAWTTest.java

```diff
@@ -136,40 +157,36 @@ public void compareText (boolean mustEqual) {
         }
     }
 
-    public void doTest() throws Exception {
+    private static void test() throws Exception {
         getPrimaryClipboard();
-        ExtendedRobot robot = new ExtendedRobot();
-        robot.waitForIdle(1000);
-        frame.setLocation(100, 100);
-        robot.waitForIdle(1000);
-
-        if (pClip != null) {
-            Point ttf1Center = tf1.getLocationOnScreen();
-            ttf1Center.translate(tf1.getWidth()/2, tf1.getHeight()/2);
-
-            robot.glide(new Point(0, 0), ttf1Center);
-            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
-            robot.waitForIdle(20);
-            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
-            robot.waitForIdle(20);
-            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
-            robot.waitForIdle(20);
-            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
-            robot.waitForIdle(2000);
-
-            getClipboardsContent();
-            compareText(true);
-
-            //Change the text selection to update the contents of primary clipboard
-            robot.mouseMove(ttf1Center);
-            robot.mousePress(MouseEvent.BUTTON1_MASK);
-            robot.delay(200);
-            robot.mouseMove(ttf1Center.x + 15, ttf1Center.y);
-            robot.mouseRelease(MouseEvent.BUTTON1_MASK);
-            robot.waitForIdle(2000);
-
-            getClipboardsContent();
-            compareText(false);
-        }
+        robot.waitForIdle(500);
+
+        EventQueue.invokeAndWait(() -> {
+            Point center = tf1.getLocationOnScreen();
+            center.translate(tf1.getWidth() / 2, tf1.getHeight() / 2);
+            ttf1Center = center;
+        });
+
+        robot.glide(new Point(0, 0), ttf1Center);
```

Comment:
Gliding from `0,0` can sometimes be time consuming, e.g. on my `3440x1440` screen this test took 24s, but with the following patch only 7s.

```patch
--- a/test/jdk/java/awt/datatransfer/Independence/IndependenceAWTTest.java
+++ b/test/jdk/java/awt/datatransfer/Independence/IndependenceAWTTest.java
@@ -59,6 +59,7 @@ public class IndependenceAWTTest {
     private static Clipboard primaryClip;
     private static ExtendedRobot robot;
     private static volatile Point ttf1Center;
+    private static volatile Point glideStartLocation;
 
     public static void main (String[] args) throws Exception {
         try {
@@ -165,9 +166,11 @@ private static void test() throws Exception {
             Point center = tf1.getLocationOnScreen();
             center.translate(tf1.getWidth() / 2, tf1.getHeight() / 2);
             ttf1Center = center;
+            glideStartLocation = frame.getLocationOnScreen();
+            glideStartLocation.x -= 10;
         });
 
-        robot.glide(new Point(0, 0), ttf1Center);
+        robot.glide(glideStartLocation, ttf1Center);
         robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
         robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
         robot.waitForIdle(20);
```

Reply:
Thanks. With your patch the test executes much faster. Since I moved the frame to center, the time required for robot.glide() was more noticeable and using `glideStartLocation` works well.